src/compiler.ts

```diff
@@ -1084,8 +1084,11 @@ export class Compiler extends DiagnosticEmitter {
               (<EnumValue>member).identifierNode.range.atEnd
             );
           }
+          var previousValueExpression = isInline ?
+            module.i32(changetype<i32>(previousValue.constantIntegerValue)) :
```

Comment:
This works under the assumption that inline enums are immutable- this is an assertion that I think makes sense but I lack context

Reply:
That's the right location and idea, yeah. Note that `previousValue.constantIntegerValue` is a portable I64 here, intended to be accessed via the `i64_*` helpers (it's a Long.js instance in JS, an i64 in Wasm):

```ts
var previousValueExpr: ExpressionRef;
if (isInline) {
  let value = previousValue.constantIntegerValue;
  assert(!i64_high(value));
  previousValueExpr = module.i32(i64_low(value));
} else {
  previousValueExpr = module.global_get(previousValue.internalName, NativeType.I32);
}
```

Should work fine then :)